pandas/core/frame.py

```diff
@@ -866,6 +866,12 @@ def dot(self, other):
         else:  # pragma: no cover
             raise TypeError('unsupported type: %s' % type(other))
 
+    def __matmul__(self, other):
+        dot_types = (Index, DataFrame, Series, np.ndarray, list)
```

Comment:
I feel like this should become a common resource, since you use it in multiple places.